### PR TITLE
Issue #23 Backend File Structure

### LINKED ISSUE #23: Backend File Structure
As a developer, I want a file structure that is logical and easy to navigate. 

**Acceptance Criteria**

- [x] The file structure follows a Service/Repository/Controller model 
- [x] The file structure follows a Enterprise Java model

### PR DESCRIPTION
**Pull request by MCoriale**
Fixes backend file structure to incorporate controllers, repositories, and services. 
Also added new classes (just templates) for the current DB model we have right now, those are not completed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ build/
 !gradle/wrapper/gradle-wrapper.jar
 !**/src/main/**/build/
 !**/src/test/**/build/
+!.metadata
```
Comment (DRSisco) [inline .gitignore]: I believe `!` is unignore or considered an "exception"
If you want to ignore .metadata, I think you want to just simply put `.metadata`.  Actually this is a directory so we should probably do `.metadata/`?

True for stuff above this line too if that's the intent.

https://git-scm.com/docs/gitignore#_pattern_format
>An optional prefix "`!`" which negates the pattern; any matching file excluded by a previous pattern will become included again. It is not possible to re-include a file if a parent directory of that file is excluded. Git doesn’t list excluded directories for performance reasons, so any patterns on contained files have no effect, no matter where they are defined. Put a backslash ("`\`") in front of the first "`!`" for patterns that begin with a literal "`!`", for example, "`\!important!.txt`".

File: src/main/java/edu/rosehulman/covidtracer/covidtracerapplication/repository/PersonRepository.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package edu.rosehulman.covidtracer.covidtracerapplication.repository;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import java.util.Optional;
+
+public class PersonRepository<Person, ID> implements PagingAndSortingRepository<Person, ID> {
```
Comment (DRSisco) [inline src/main/java/edu/rosehulman/covidtracer/covidtracerapplication/repository/PersonRepository.java:10]: Thanks for skeletonizing all this!

**Review by DRSisco — COMMENTED** [review]
🥳 Everyone is requested to review! 🥳

**Comment by DRSisco** [comment]
In the Pull Request description can we add either `closes #23` (Will auto-close on merge to `main`) or `#23` (simply references) to link this PR back to the issue, that way when you look at either the PR or the issue you can quickly find the other. 

Thanks!

Edit: I see you did link it with the github UI directly. That works too. Don't mind me 😅

**Comment by MCoriale** [comment]
Closes #24

**Review by meistejw — APPROVED** [review]
Look all good to me. Nice structure to work off of.

**Pull request merged**